lib/vault.py

```diff
@@ -115,34 +115,65 @@ def initialize(self, secrets = 5, threashold = 3):
         """
 
         client = self.connect()
-        if client.is_initialized():
+        if client.sys.is_initialized():
             print("Vault is already initialized")
-            if client.is_sealed():
+            if client.sys.is_sealed():
                 print("Unsealing Vault")
                 self.unseal()
             else:
                 print("Vault already unsealed")
         else:
             print("Initializing with {} secrets and {} needed to unseal".format(secrets, threashold))
-            result = client.initialize(secrets, threashold)
+            result = client.sys.initialize(secret_shares=1,
+                                           secret_threshold=1,
+                                           stored_shares=1,
+                                           recovery_shares=secrets,
+                                           recovery_threshold=threashold)
 
             token_file = self.path(VAULT_TOKEN)
             key_file = self.path(VAULT_KEY)
             with open(token_file, "w") as fh:
                 fh.write(result["root_token"])
             for i in range(secrets):
                 with open(key_file + str(i+1), "w") as fh:
-                    fh.write(result["keys"][i])
+                    fh.write(result["recovery_keys"][i])
 
+            # DP TODO: refactor code so that the root token is revoked after configuration?
+            # DP ???: If no root token, how to auth for populating future values?
             print()
-            print("======== WARNING WARNING WARNING ========")
-            print("= Vault root token and unseal keys were =")
-            print("= written to disk. PROTECT these files. =")
-            print("======== WARNING WARNING WARNING ========")
-
+            print("========= WARNING WARNING WARNING =========")
+            print("= Vault root token and recovery keys were =")
+            print("= written to disk. PROTECT these files.   =")
+            print("========= WARNING WARNING WARNING =========")
             print()
-            print("Unsealing Vault")
-            client.unseal_multi(result["keys"])
+
+        # DP NOTE: When using the DynamoDB backend it is common for right after
+        #          initializing for requests to Vault to response with the error
+        #          > local node not active but active cluster node not found <
+        #          If given a little bit of time Vault will respond successfully
+        #          to requests
+
+        def poll():
+            """Check to see if Vault responds to a request without an error"""
+            try:
+                self.connect(VAULT_TOKEN).sys.list_enabled_audit_devices()
+                return True
+            except:
```

Comment:
Do we want to catch specific exceptions so we can fail faster if something unexpected occurs?

Reply:
Updated with both catching the exception and checking the exception message